ignite/pkg/cosmosgen/generate.go

```diff
@@ -117,28 +118,35 @@ func (g *generator) setup() (err error) {
 			if err != nil {
 				return err
 			}
-			includes := []string{}
-			if len(modules) > 0 {
 
-				includes, err = g.resolveIncludes(path) // For versioning issues, we do dependency/includes resolution per module
+			var includes []string
+			if len(modules) > 0 {
+				// For versioning issues, we do dependency/includes resolution per module
+				includes, err = g.resolveIncludes(path)
 				if err != nil {
 					return err
 				}
 			}
+
 			modulesInPath = ModulesInPath{
-				Path: path, // Each go module
-				ModuleIncludes: ModuleIncludes{ // Has a set of sdk modules and a set of includes to build them
-					Includes: includes,
-					Modules:  modules,
-				},
+				Path:     path,
+				Modules:  modules,
+				Includes: includes,
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = modulesInPath.ModuleIncludes
+		g.thirdModules[modulesInPath.Path] = append(
+			g.thirdModules[modulesInPath.Path],
+			modulesInPath.Modules...,
+		)
+		g.thirdModuleIncludes[modulesInPath.Path] = append(
```

Comment:
What is the difference between both variables? And why this new variable `thirdModuleIncludes` was introduced?

Reply:
The `thirdModules` is analog to `appModules` and it contains a map between Go dependencies and proto module analysis information for app dependencies.

The `thirdModuleIncludes` is analog to `appIncludes` and it contains a map between app dependencies and `protoc` include paths required for each dependency. These include paths are used to generate Typescript code. These paths might contain the path to Buf extracted proto files if the dependency uses Buf.

The idea of having both `third...` was to be consistent with the target branch implementation which brings back the original module discovery implementation and uses `app...`.